### PR TITLE
Disallow attributes on type extensions

### PR DESCRIPTION
**Pull request by nelson-wu**
Fixes https://github.com/dotnet/fsharp/issues/7394

Before, when attributes get added to type extensions, they can be parsed, no errors are thrown, but the attributes are silently dropped inside the type checker. This PR explicitly disallows that.

### CODE REVIEW

**Comment by nelson-wu** [comment]
@KevinRansom @dsyme Could you guys take a look at this PR?

**Comment by KevinRansom** [comment]
@nelson, can you change the error message per @dsyme, then I will merge the change.

Thanks

Kevin

**Comment by nelson-wu** [comment]
@KevinRansom @dsyme Thanks for the review! I've made the changes

File: tests/fsharp/Compiler/Language/TypeAttributeTests.fs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+open System
+
+[<NoEquality>]
+type String with 
+    member this.test = 42
+            """
+            FSharpErrorSeverity.Error
+            3245
+            (4, 1, 4, 15)
+            "Attributes cannot be assigned to type extensions."
```
Comment (cartermp) [inline tests/fsharp/Compiler/Language/TypeAttributeTests.fs]: Needs update

Comment (nelson-wu) [reply tests/fsharp/Compiler/Language/TypeAttributeTests.fs]: Oops 😛

**Comment by KevinRansom** [comment]
@nelson-wu, thanks mate, when it goes green, I'll merge it.

File: src/fsharp/TypeChecker.fs [hunk 2]
```diff
@@ -16429,6 +16429,11 @@ module TcDeclarations =
                 if not (isNil members) && tcref.IsTypeAbbrev then 
                     errorR(Error(FSComp.SR.tcTypeAbbreviationsCannotHaveAugmentations(), tyDeclRange))
 
+                let (ComponentInfo (attributes, _, _, _, _, _, _, _)) = synTyconInfo
+                if not (List.isEmpty attributes) && (declKind = ExtrinsicExtensionBinding || declKind = IntrinsicExtensionBinding) then
```
Comment (TIHan) [inline src/fsharp/TypeChecker.fs]: Great work!

**Pull request merged**